targets/alert-cli/src/index.js

```diff
@@ -199,110 +204,89 @@ async function getNewerTagsFromRepo(repository, lastTag) {
 
 /**
  *
- * @param {alerts.GitTagData[]} tags include the last tag from the previous run
- * @param {string} repositoryId
- * @param {string[]} ficheVddIDs
+ * @param {string} repository
+ * @param {alerts.GitTagData} currentTag first tag to compute diff (most recent)
+ * @param {alerts.GitTagData} previousTag second tag to compute diff (least recent)
+ * @returns {Promise<alerts.AlertChanges[]>}
  */
-async function getDiffFromTags(tags, repositoryId, ficheVddIDs) {
-  let [previousTag] = tags;
-  const [, ...newTags] = tags;
+async function getAlertChangesFromTags(repository, currentTag, previousTag) {
+  const fileFilter = await getFileFilter(repository);
+  const diffProcessor = getDiffProcessor(repository);
 
-  /** @type {(alerts.AlertChanges)[]}  */
-  const allChanges = [];
+  const currentCommit = currentTag.commit;
+  const previousCommit = previousTag.commit;
+  const [prevTree, currTree] = await Promise.all([
+    previousCommit.getTree(),
+    currentCommit.getTree(),
+  ]);
+  const diff = await currTree.diff(prevTree);
+  const patches = await diff.patches();
 
-  const fileFilter = getFileFilter(repositoryId, ficheVddIDs);
-  const diffProcessor = getDiffProcessor(repositoryId);
+  const files = patches.map(getFilename).filter(fileFilter);
```

Comment:
I think we can inline this call :
```js
const files = await currTree.diff(prevTree).patches().map(getFilename).filter(fileFilter);
```
I think it's more readable because methods are self-explanatory.

Reply:
since patches() method is async we need to add parenthesis before map / filter
```js
const files = (await currTree.diff(prevTree).patches()).map(getFilename).filter(fileFilter);
```

Reply:
Yes, indeed. It's more ugly with await. Perhaps two lines :
```js
const patches = await currTree.diff(prevTree).patches()
const modifiedFiles = patches.map(getFilename).filter(fileFilter);
```
What do you think?

Reply:
diff() method is async too
```js
const patches = await (await currTree.diff(prevTree)).patches()
const modifiedFiles = patches.map(getFilename).filter(fileFilter);
```

Reply:
🤦 Ok, I'll go to the bed, I'll see you tomorrow 😄 (I close this discussion).